src/Simplex/FileTransfer/Server/Store.hs

```diff
@@ -1 +1,99 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+{-# HLINT ignore "Use lambda-case" #-}
+
 module Simplex.FileTransfer.Server.Store where
+
+import Simplex.Messaging.Encoding.String
+import Simplex.Messaging.Protocol hiding (SParty, SRecipient, SSender)
+import Control.Monad
+import Data.Functor (($>))
+import Simplex.Messaging.TMap (TMap)
+import qualified Simplex.Messaging.TMap as TM
+import Simplex.Messaging.Util (ifM, ($>>=))
+import UnliftIO.STM
+import Simplex.FileTransfer.Protocol (SFileParty (..))
+import Data.List.NonEmpty (NonEmpty)
+import Data.Aeson.KeyMap (mapMaybe)
+import Data.Maybe (fromMaybe)
+
+data FileStore = FileStore
+  { chunks :: TMap SenderId FileRec,
+    recipients :: TMap RecipientId (SenderId, RcvPublicVerifyKey)
+  }
+
+data FileRec = FileRec
+  { senderId :: SenderId,
+    senderKey :: SndPublicVerifyKey,
+    readers :: TVar [RecipientId],
+    filepath :: String
```

Comment:
```suggestion
    filePath :: TVar (Maybe FilePath)
```

It's created when there is no file yet, file is added with another command. FilePath is a type synonym for String.

Reply:
which means there will be one more function to add this path.